lib/vfs/src/host_fs.rs

```diff
@@ -533,76 +525,156 @@ impl VirtualFile for Stdout {
         Ok(())
     }
 
-    fn bytes_available(&self) -> Result<usize> {
-        host_file_bytes_available(io::stdout().try_into_filedescriptor()?)
+    fn get_special_fd(&self) -> Option<u32> {
+        Some(1)
     }
 
-    fn get_fd(&self) -> Option<FileDescriptor> {
-        io::stdout().try_into_filedescriptor().ok()
+    fn poll_read_ready(self: Pin<&mut Self>, _cx: &mut Context<'_>) -> Poll<io::Result<usize>> {
+        Poll::Ready(Ok(0))
     }
-}
 
-/// A wrapper type around Stderr that implements `VirtualFile` and
-/// `Serialize` + `Deserialize`.
-#[derive(Debug, Default)]
-#[cfg_attr(feature = "enable-serde", derive(Serialize, Deserialize))]
-pub struct Stderr;
+    fn poll_write_ready(self: Pin<&mut Self>, _cx: &mut Context<'_>) -> Poll<io::Result<usize>> {
+        Poll::Ready(Ok(8192))
```

Comment:
Why this number and not other? Should it be a const?

Reply:
it should be const indeed

as for why this exact number itself - it should be aligned with a page size (4K) (for better caching performance) and be of reasonable size. network stacks have shown that the size of modern packets is too small (1500bytes) for a decent throughput thus quite often a switch to jumbo frames (9000 bytes packets) can have quite a significant improvement on bandwidth - a similar IO overhead will occur on piped data too however without testing its basically an educated guess.

given most code i've seen (like `tokio` ) just ignores the poll result numbers and hits the file descriptor with non blocking writes until it returns `E_AGAIN` the use of the number is hard measure - still its better than putting 1 as there could be nasty edge cases if one actually does put some logic on that number (for instance, wait for at least 32bits before writing a length prefix on the pipe).

either way another educated guess is also welcome for this number and it indeed should be a const.

Reply:
Factored it out to a const and added a doc comment.